### PR TITLE
sim/nsh: Configure gcov to be enabled by default

### PR DESCRIPTION
**Pull request by W-M-R**
*Note: Please adhere to [Contributing Guidelines](https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md).*

## Summary

sim/nsh: Configure gcov to be enabled by default

## Impact

*Update this section, where applicable, on how change affects users,
 build process, hardware, documentation, security, compatibility, etc.*

## Testing

*Update this section with details on how did you verify the change,
 what Host was used for build (OS, CPU, compiler, ..), what Target was
 used for verification (arch, board:config, ..), etc. Providing build
 and runtime logs from before and after change is highly appreciated.*

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/sim/sim/sim/configs/nsh/defconfig [hunk 1]
```diff
@@ -16,6 +16,8 @@ CONFIG_BOARDCTL_POWEROFF=y
 CONFIG_BOARD_LOOPSPERMSEC=0
 CONFIG_BOOT_RUNFROMEXTSRAM=y
 CONFIG_BUILTIN=y
+CONFIG_COVERAGE_ALL=y
+CONFIG_COVERAGE_TOOLCHAIN=y
```
Comment (anchao) [inline boards/sim/sim/sim/configs/nsh/defconfig:20]: Why not add a new config?

Comment (xiaoxiang781216) [reply boards/sim/sim/sim/configs/nsh/defconfig:20]: to save the ci resource. @W-M-R let's move to configs/asan.

Comment (anchao) [reply boards/sim/sim/sim/configs/nsh/defconfig:20]: sim/kasan already has this config

Comment (W-M-R) [reply boards/sim/sim/sim/configs/nsh/defconfig:20]: okey，My original intention was that sim kasan was unable to directly use this gcov configuration due to the setting of initargs. So I want to make it easier to open the image of NSH

**Pull request closed without merging**